test/SpecHelper.hs

```diff
@@ -46,16 +45,19 @@ validateOpenApiResponse headers = do
     respHeaders `shouldSatisfy`
```

Comment:
Not really related to the PR, but could these checks be replaced by a single `shouldRespondWith` ?

Reply:
My understanding is that `shouldRespondWith` requires specifying the full response body, which is why it's avoided here.